packages/core/src/tools/sanitize.ts

```diff
@@ -0,0 +1,235 @@
+import type { Context, ContextArray, ContextValue } from './context'
+import type { ObjectWithToJsonMethod } from './utils'
+import { detachToJsonMethod, ONE_KIBI_BYTE } from './utils'
+
+type ContainerElementsToProcess = { source: ContextArray | Context; target: ContextArray | Context; path: string }
+// eslint-disable-next-line @typescript-eslint/ban-types
+type ExtendedContextValue = ContextValue | symbol | bigint | Function
+
+// The maximum size of a single event is 256KiB. By default, we ensure that user-provided data
+// going through sanitize fits inside our events, while leaving room for other contexts, metadata, ...
+const SANITIZE_DEFAULT_MAX_CHARACTER_COUNT = 220 * ONE_KIBI_BYTE
+
+// Symbol for the root element of the JSONPath used for visited objects
+const JSON_PATH_ROOT_ELEMENT = '$'
+
+/**
+ * Ensures user-provided data is 'safe' for the SDK
+ * - Deep clones data
+ * - Removes cyclic references
+ * - Transforms unserializable types to a string representation
+ *
+ * LIMITATIONS:
+ * - Size is in characters, not byte count
+ * - Size does not take into account indentation that can be applied to JSON.stringify
+ * - Non-numerical properties of Arrays are ignored. Same behavior as JSON.stringify
+ *
+ * @param source              User-provided data meant to be serialized using JSON.stringify
+ * @param maxCharacterCount   Maximum number of characters allowed in serialized form
+ * @returns
+ */
+export function sanitize(source: string, maxCharacterCount?: number): string | undefined
+export function sanitize(source: unknown, maxCharacterCount?: number): ContextValue
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏adding a signature matching objects could avoid some cast

```suggestion
export function sanitize(source: string, maxCharacterCount?: number): string | undefined
export function sanitize(source: object | Context, maxCharacterCount?: number): Context
export function sanitize(source: unknown, maxCharacterCount?: number): ContextValue
```

Reply:
Added  `export function sanitize(source: Context, maxCharacterCount?: number): Context`
However, `object` is not necessarily a container